src/rdkafka_int.h

```diff
@@ -492,6 +493,23 @@ struct rd_kafka_s {
                 rd_kafka_timer_t    txn_coord_tmr;
         } rk_eos;
 
+
+        /**
+         * Consumer state
+         *
+         * @locality rdkafka main thread
+         * @locks_required none
+         */
+        struct {
+                /** Application consumer queue for messages, events and errors.
+                 *  (typically points to rkcg_q) */
+                rd_kafka_q_t *q;
+                /** Current assigned partitions through assign() et.al. */
+                rd_kafka_assignment_t assignment;
+                /** Waiting for this number of commits to finish. */
+                int wait_commit_cnt;
+        } rk_consumer;
```

Comment:
yes good... `rk_cgrp` should be in here too, and the structure is inside out since `rk` should be a member of `rk_consumer`, but I like that assignment is differentiated from group assignment.